Adding a speed button to android auto

AwsomeFox opened the pull request:
Possible start for issue #4. 

Doesn't have any tests and is kind of janky selecting the next speed but it works. Figured I'd open a draft pr for discussion. If it works for you I can work on polishing it.

AwsomeFox committed:
Adding a speed button to android auto

mattttvaughn commented on app/src/main/java/io/github/mattpvaughn/chronicle/features/player/CustomActions.kt:
```diff
@@ -10,25 +10,44 @@ import com.google.android.exoplayer2.ControlDispatcher
 
 }
 
+fun changeSpeed(prefsRepo: PrefsRepo) {
```
Good stuff 👍

mattttvaughn commented on build.gradle:
```diff
@@ -12,7 +12,7 @@ buildscript {
         okhttpVersion = '4.9.1'
         dataBindingCompilerVersion = gradleVersion // Always need to be the same.
-        roomVersion = "2.2.6"
+        roomVersion = "2.3.0"
```
I'd prefer to keep dependency upgrades in self-contained PRs for testing and ability to roll back easily

AwsomeFox replied:
Yeah I agree with that I just couldn't build it without that version. I'll roll it back.

mattttvaughn commented on app/src/main/java/io/github/mattpvaughn/chronicle/features/player/AudiobookMediaSessionCallback.kt:
```diff
@@ -122,6 +123,13 @@ class AudiobookMediaSessionCallback @Inject constructor(
+        changeSpeed(prefsRepo)
+        Timber.i("New Speed: %s", prefsRepo.playbackSpeed)
+        Toast.makeText(appContext, "New Speed: " + prefsRepo.playbackSpeed, Toast.LENGTH_SHORT)
+            .show()
```
Is there another way we could convey playback speed to users? I know the screenshot posted for on [the issue](https://github.com/mattttvaughn/chronicle/issues/4) shows the playback speed on the icon, not sure about the difficulty of that

Alternatively, if there isn't an easy way to do the above, using a fast forward icon instead could be a bit more intuitive for a user who we shouldn't expect to read a 5-10s toast. I'd love to hear y'all thoughts on this

AwsomeFox replied:
Yeah I gave it a little bit of time to try and figure that out. I added the button since I'm heading on a road trip and wanted the option. When I get back I'll probably do some more digging to try and figure it out.

mattttvaughn reviewed (commented):
Excellent work, thanks for contributing! 

I think we should think out a way to communicate playback speed without toasts, but the core of this commit is solid imo

AwsomeFox committed:
reverting version increase, adding current playback speed to button in android auto, reverting enabling android auto.

AwsomeFox committed:
removing now unused icon

AwsomeFox commented:
Added the suggested button displaying current speed. It will update to the new speed on press. Also reverted the room version updates and the enabling of android auto. I think this is ready for a real review now. @mattttvaughn

mattttvaughn commented on app/src/main/java/io/github/mattpvaughn/chronicle/features/player/CustomActions.kt:
```diff
@@ -46,6 +52,13 @@ fun changeSpeed(prefsRepo: PrefsRepo) {
         3.0f -> prefsRepo.playbackSpeed = 0.5f
         else -> prefsRepo.playbackSpeed = 1.0f
     }
+    mediaSessionConnector.setCustomActionProviders(
```
good solution for updating the icon!

mattttvaughn reviewed (approved):
Really great stuff! Appreciate the contribution :+1:
